test/Microsoft.Sbom.Tool.Tests/IntegrationTests.cs

```diff
@@ -0,0 +1,219 @@
+    {
+        var arguments = $"generate -ps IntegrationTests -pn IntegrationTests -pv 1.2.3 -m \"{testFolderPath}\"  -b . -bc \"{GetSolutionFolderPath()}\"";
+
+        var (stdout, stderr, exitCode) = LaunchAndCaptureOutput(arguments);
+
+        Assert.AreEqual(stderr, string.Empty);
+        var manifestFolderPath = AppendFullManifestFolderPath(testFolderPath);
+        var jsonFilePath = Path.Combine(manifestFolderPath, ManifestFileName);
+        var shaFilePath = Path.Combine(manifestFolderPath, "manifest.spdx.json.sha256");
+        Assert.IsTrue(File.Exists(jsonFilePath));
+        Assert.IsTrue(File.Exists(shaFilePath));
+        Assert.AreEqual(0, exitCode.Value);
+    }
+
+    private string CreateTestFolder()
+    {
+        var testFolderPath = Path.GetFullPath(Path.Combine(testContext.ResultsDirectory, testContext.TestName));
+        Directory.CreateDirectory(testFolderPath);
+        return testFolderPath;
+    }
+
+    private static string AppendFullManifestFolderPath(string manifestDir)
+    {
+        return Path.Combine(manifestDir, ManifestRootFolderName, "spdx_2.2");
+    }
+
+    private static string GetSolutionFolderPath()
+    {
+        return Path.GetFullPath(Path.Combine(Assembly.GetExecutingAssembly().Location, "..", "..", ".."));
+    }
+
+    private static string GetAppName()
+    {
+        return IsWindows ? "Microsoft.Sbom.Tool.exe" : "Microsoft.Sbom.Tool";
+    }
+
+    private static (string stdout, string stderr, int? exitCode) LaunchAndCaptureOutput(string? arguments)
+    {
+        var stdout = string.Empty;
+        var stderr = string.Empty;
+        int? exitCode = null;
+        Process process = null;
+
+        try
+        {
+            process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = GetAppName(),
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    Arguments = arguments ?? string.Empty,
+                }
+            };
+
+            process.OutputDataReceived += (sender, e) =>
+            {
+                if (!string.IsNullOrEmpty(e.Data))
+                {
+                    stdout += e.Data + Environment.NewLine;
+                }
+            };
+
+            process.ErrorDataReceived += (sender, e) =>
+            {
+                if (!string.IsNullOrEmpty(e.Data))
+                {
+                    stderr += e.Data + Environment.NewLine;
+                }
+            };
+
+            process.Start();
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
+        }
+        catch (Exception ex)
+        {
+            // Handle any exceptions here
+            Console.WriteLine("Error: " + ex.Message);
```

Comment:
I would expect the test to fail on an exception- why would we not want that to be the case?

Reply:
Good point. I wrote this via Copilot and wasn't thinking in terms of errors occurring here. I _think_ upstream tests would fail, but an `Assert.Fail` here will guarantee it.